examples/Analytic_300hPa_Trough.py

```diff
@@ -0,0 +1,179 @@
+T = single_300hPa_trough(parameter='temp')
+
+# Lifting Trough
+# Z = lifting_300hPa_trough(parameter='hght')
+# T = lifting_300hPa_trough(parameter='temp')
+
+# Digging Trough
+# Z = digging_300hPa_trough(parameter='hght')
+# T = digging_300hPa_trough(parameter='temp')
+
+
+######################################################################
+# Set geographic parameters for analytic grid to then
+#
+
+lats = np.linspace(35, 50, 101)
+lons = np.linspace(260, 290, 101)
+lon, lat = np.meshgrid(lons, lats)
+
+# Calculate Geostrophic Wind from Analytic Heights
+f = mpcalc.coriolis_parameter(lat * units('degrees'))
+dx, dy = mpcalc.lat_lon_grid_deltas(lons, lats)
+ugeo, vgeo = mpcalc.geostrophic_wind(Z*units.meter, f, dx, dy, dim_order='yx')
+
+# Get the wind direction for each point
+wdir = mpcalc.wind_direction(ugeo, vgeo)
+
+# Compute the Gradient Wind via an approximation
+dydx = mpcalc.first_derivative(Z, delta=dx, axis=1)
+d2ydx2 = mpcalc.first_derivative(dydx, delta=dx, axis=1)
+R = ((1+dydx.m**2)**(3./2.))/d2ydx2.m
+
+geo_mag = mpcalc.wind_speed(ugeo, vgeo)
+grad_mag = geo_mag.m - (geo_mag.m**2)/(f.magnitude*R)
+
+ugrad, vgrad = mpcalc.wind_components(grad_mag*units('m/s'), wdir)
+
+# Calculate Ageostrophic wind
+uageo = ugrad - ugeo
+vageo = vgrad - vgeo
+
+# Compute QVectors
+uqvect, vqvect = mpcalc.q_vector(ugeo, vgeo, T*units.degC, 500*units.hPa, dx, dy)
+
+# Calculate divergence of the ageostrophic wind
+div = mpcalc.divergence(uageo, vageo, dx, dy, dim_order='yx')
+
+# Calculate Relative Vorticity Advection
+relvor = mpcalc.vorticity(ugeo, vgeo, dx, dy, dim_order='yx')
+adv = mpcalc.advection(relvor, (ugeo, vgeo), (dx, dy), dim_order='yx')
+
+
+######################################################################
+# Create figure containing Geopotential Heights, Temperature, Divergence
+# of the Ageostrophic Wind, Relative Vorticity Advection (shaded),
+# geostrphic wind barbs, and Q-vectors.
+#
+
+fig = plt.figure(figsize=(10, 10))
+ax = plt.subplot(111)
+
+# Plot Geopotential Height Contours
+cs = ax.contour(lons, lats, Z, range(0, 12000, 120), colors='k')
+plt.clabel(cs, fmt='%d')
+
+# Plot Temperature Contours
+cs2 = ax.contour(lons, lats, T, range(-50, 50, 2), colors='r', linestyles='dashed')
+plt.clabel(cs2, fmt='%d')
+
+# Plot Divergence of Ageo Wind Contours
+cs3 = ax.contour(lons, lats, div*10**9, np.arange(-25, 26, 3), colors='grey',
+                 linestyles='dotted')
+plt.clabel(cs3, fmt='%d')
+
+# Plot Rel. Vor. Adv. colorfilled
+cf = ax.contourf(lons, lats, adv*10**9, np.arange(-20, 21, 1), cmap=plt.cm.bwr)
+cbar = plt.colorbar(cf, orientation='horizontal', pad=0.05, aspect=50)
+cbar.set_label('Rel. Vor. Adv.')
+
+# Plot Geostrophic Wind Barbs
+wind_slice = slice(5, None, 10)
+ax.barbs(lons[wind_slice], lats[wind_slice],
+         ugeo[wind_slice, wind_slice].to('kt').m, vgeo[wind_slice, wind_slice].to('kt').m)
+
+# Plot Ageostrophic Wind Vectors
+# qvec_slice = slice(None, None, 10)
```

Comment:
Why have this commented out?

Reply:
Too many things going on in the plot, can take it out, but its another nice version instead of plotting the Q-vectors.